File: DiabetesManagement.Api/v3/Inventory/Models/Inventory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DiabetesManagement.Models
{
    [MessagePack.MessagePackObject(true),
     Table(nameof(Inventory))]
    public class Inventory
    {
        [Key]
        public Guid InventoryId { get; set; }
        public Guid UserId { get; set; }
        public string? Subject { get; set; }
        public string? DefaultIntent { get; set; }
        public string? Hash { get; set; }
        public DateTimeOffset Created { get; set; }
        public DateTimeOffset? Modified { get; set; }
    }
}

File: v3/webapp/src/models/inventory.ts
import { IResponse } from "./response";

export interface IInventory {
    inventoryId: string;
    userId: string;
    subject: string;
    defaultIntent: string;
    hash: string;
    created: Date;
    modified: Date;
}

export interface IInventoryHistory {
    inventoryHistoryId: string;
    inventoryId: string;
    version: number;
    intent: string;
    value: string;
    created: Date;
    hash: string;
    inventory: IInventory
}

export interface IInventoryRequest {
    items: string;
    key: string;
    userId: string;
    type: string;
}

export interface IInventoryService {
    save(request: IInventoryRequest): Promise<IResponse<IInventoryHistory>>;
    get(request: IInventoryRequest) : Promise<IResponse<IInventoryHistory>>;
}